### PR TITLE
Fixing minor issues

### PR DESCRIPTION
**Pull request by shrikant1407**
Fixing minor issues
- Add certain checks at various situations and update usage
- Removed unused code
- Fix memory leaks

Signed-off-by: Shrikant Temburwar <shrikant.temburwar@intel.com>

### CODE REVIEW

File: device_modules/fdo_sys/sys_utils_linux.c [hunk 1]
```diff
@@ -347,7 +347,13 @@ size_t get_file_sz(char const *filename)
 	FILE *fp = fopen(filename, "rb");
```
Comment (vikashmishra07) [inline device_modules/fdo_sys/sys_utils_linux.c:348]: use fopen_s

File: lib/fdotypes.c [hunk 2]
```diff
@@ -1364,8 +1337,7 @@ fdo_public_key_t *fdo_public_key_read(fdor_t *fdor)
 	case FDO_CRYPTO_PUB_KEY_ENCODING_X5CHAIN:
 		LOG(LOG_ERROR, "Invalid PublicKey: pkEnc.X5CHAIN is not supported.\n");
 		goto err;
-	case FDO_CRYPTO_PUB_KEY_ENCODING_COSEKEY:
-		;
+	case FDO_CRYPTO_PUB_KEY_ENCODING_COSEKEY: ;
```
Comment (vikashmishra07) [inline lib/fdotypes.c:1340]: Why you need ";"

**Review by vikashmishra07 — COMMENTED** [review]
Refactoring of sys and restinterface is not done.
Are you planning to do in next release

File: lib/fdotypes.c (hunk 2 above)
Comment (shrikant1407) [reply lib/fdotypes.c:1340]: Labels (FDO_CRYPTO_PUB_KEY_ENCODING_COSEKEY) can only be followed by statements, and declarations do not count as statements in C. So ';' is used to make an empty statement here.

File: device_modules/fdo_sys/sys_utils_linux.c (hunk 1 above)
Comment (shrikant1407) [reply device_modules/fdo_sys/sys_utils_linux.c:348]: fopen_s is only supported with the C11 standard. Compiling with the c11 flag produces more errors. Will be fixed in the next release.

**Comment by shrikant1407** [comment]
> Refactoring of sys and restinterface is not done. Are you planning to do in next release

Yes, refactoring will be done in the next release

**Review by KiranSukhavasi — APPROVED** [review]

**Review by vikashmishra07 — APPROVED** [review]

**Review by himanshu7196 — APPROVED** [review]

**Pull request merged**